libs/dotnet-sdk/State/FeatureBoardState.cs

```diff
@@ -9,50 +9,49 @@ public class FeatureBoardState : IFeatureBoardState
   public DateTimeOffset? LastUpdated { get; private set; }
   public DateTimeOffset? LastModified { get; private set; }
   private readonly IFeatureBoardExternalState? _externalState;
-
   private static readonly ConcurrentDictionary<string, FeatureConfiguration> _cache = new(StringComparer.OrdinalIgnoreCase);
 
-  public FeatureBoardState(IFeatureBoardExternalState? externalState)
+  public FeatureBoardState(IFeatureBoardExternalState? externalState = null)
   {
     _externalState = externalState;
   }
 
   public FeatureBoardStateSnapshot GetSnapshot() => new(_cache);
 
-  public async Task InitialiseState(Dictionary<string, FeatureConfiguration>? features, DateTimeOffset? lastModified, CancellationToken cancellationToken)
+  public async Task InitialiseState(List<FeatureConfiguration>? features, DateTimeOffset? lastModified, CancellationToken cancellationToken)
   {
     if (features is null && _externalState is not null)
     {
       features = await _externalState.GetState(cancellationToken);
     }
 
-    foreach (var featureConfiguration in features ?? new Dictionary<string, FeatureConfiguration>())
+    foreach (var feature in features ?? new List<FeatureConfiguration>())
```

Comment:
This won't remove items which have been removed from the passed in list?

Reply:
Possible this method needs to throw if it's already been initialised?

Reply:
Fixed